epictrack-api/src/api/services/work.py

```diff
@@ -674,3 +583,96 @@ def check_work_nation_existence(
         if query.count() > 0:
             return True
         return False
+
+    @classmethod
+    def handle_phase(cls, work_phase, phase_event_templates):  # pylint: disable=too-many-locals
+        """Create a new work phase and related events and event configuration entries"""
+        work_phase = WorkPhase.flush(WorkPhase(**work_phase))
+        event_configurations = []
+        for parent_config in list(
+            filter(lambda x: not x["parent_id"], phase_event_templates)
```

Comment:
some top level method could be looking like
```
@classmethod
def handle_phase(cls, work_phase, phase_event_templates):
    """Create a new work phase and related events and event configuration entries."""
    work_phase = WorkPhase.flush(WorkPhase(**work_phase))
    event_configurations = cls.create_event_configurations(work_phase, phase_event_templates)
    cls.create_events(work_phase, event_configurations)
```
  Again i might be simplifying it and we might need some real effort to split it .. but on the long run ,it will be worth it..